### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'status_code'

### ISSUE THREAD

**Issue by ZuluWhiskey** [PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor#129]
Apologies if this has already been raised (I've searched other issues and couldn't find this error) or if this is just me being an idiot but I had the 'camera' working fine but I'm now seeing the below errors - Any ideas what might be causing this?

**Errors:**
```
2021-06-14 09:38:07 WARNING (MainThread) [homeassistant.components.camera] Updating xiaomi_cloud_map_extractor camera took longer than the scheduled update interval 0:00:05
2021-06-14 09:38:08 ERROR (MainThread) [homeassistant.helpers.entity] Update for camera.roborock_s5_max fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 316, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 524, in async_device_update
    raise exc
  File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/xiaomi_cloud_map_extractor/camera.py", line 205, in update
    self._logged_in = self._connector.login()
  File "/config/custom_components/xiaomi_cloud_map_extractor/xiaomi_cloud_connector.py", line 104, in login
    return self.login_step_1() and self.login_step_2() and self.login_step_3()
  File "/config/custom_components/xiaomi_cloud_map_extractor/xiaomi_cloud_connector.py", line 93, in login_step_3
    return response.status_code == 200
AttributeError: 'NoneType' object has no attribute 'status_code'
```
**My config:**
```
platform: xiaomi_cloud_map_extractor
name: Roborock S5 Max
host: 192.168.1.18
token: !secret s5_max_token
username: !secret xiaomi_miio_username
password: !secret xiaomi_miio_password
country: de
#draw: ['all']
attributes:
  - calibration_points
draw:
  - charger
  - path
  - goto_path
  - obstacles
  - ignored_obstacles
  - predicted_path
  - no_go_zones
  - no_mopping_zones
  - vacuum_position
  - virtual_walls
  - zones
store_map: true
map_transformation:
  scale: 2
  trim:
    top: 30
    left: 25
    bottom: 30
    right: 30
colors:
  color_path: [255,255,255]
  color_goto_path: [0,255,0]
  color_predicted_path: [255,255,0]
  color_map_wall_v2: [0,0,0]
```
The username, password & token are all correct as I can still use the vacuum via the Xiaomi Mii integration.

Thanks in advance! 😃

**Comment by PiotrMachowski** [comment]
Can you try using [tokens extractor](https://github.com/PiotrMachowski/Xiaomi-cloud-tokens-extractor/)?

**Comment by RaveGun** [comment]
Hi,

I'm also experiencing the same issue.
I just used the python token extractor without an issue and compared the settings again.

I did worked last week and there is an old map that is loaded but there are no updates now during vacuuming.

The MiHomeApp is also reporting slow network and does not load any map.

**Comment by ZuluWhiskey** [comment]
> Can you try using [tokens extractor](https://github.com/PiotrMachowski/Xiaomi-cloud-tokens-extractor/)?

Very odd, I used that tool a week ago with zero issues but today I get errors:
```
Logging in...
Logged in.

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/urllib3/connection.py", line 159, in _new_conn
    conn = connection.create_connection(
  File "/usr/lib/python3/dist-packages/urllib3/util/connection.py", line 61, in create_connection
    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
  File "/usr/lib/python3.8/socket.py", line 918, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -2] Name or service not known

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 665, in urlopen
    httplib_response = self._make_request(
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 376, in _make_request
    self._validate_conn(conn)
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 996, in _validate_conn
    conn.connect()
  File "/usr/lib/python3/dist-packages/urllib3/connection.py", line 314, in connect
    conn = self._new_conn()
  File "/usr/lib/python3/dist-packages/urllib3/connection.py", line 171, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <urllib3.connection.VerifiedHTTPSConnection object at 0x7ff76d6c0f10>: Failed to establish a new connection: [Errno -2] Name or service not known

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/requests/adapters.py", line 439, in send
    resp = conn.urlopen(
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 719, in urlopen
    retries = retries.increment(
  File "/usr/lib/python3/dist-packages/urllib3/util/retry.py", line 436, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='de.api.io.mi.com', port=443): Max retries exceeded with url: /app/home/device_list?[[REMOVED IN CASE SENSITIVE]] (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7ff76d6c0f10>: Failed to establish a new connection: [Errno -2] Name or service not known'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "token_extractor.py", line 210, in <module>
    devices = connector.get_devices(current_server)
  File "token_extractor.py", line 102, in get_devices
    return self.execute_api_call(url, params)
  File "token_extractor.py", line 137, in execute_api_call
    response = self._session.post(url, headers=headers, cookies=cookies, params=fields)
  File "/usr/lib/python3/dist-packages/requests/sessions.py", line 581, in post
    return self.request('POST', url, data=data, json=json, **kwargs)
  File "/usr/lib/python3/dist-packages/requests/sessions.py", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/lib/python3/dist-packages/requests/sessions.py", line 646, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python3/dist-packages/requests/adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='de.api.io.mi.com', port=443): Max retries exceeded with url: /app/home/device_list?[[REMOVED IN CASE SENSITIVE]] (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7ff76d6c0f10>: Failed to establish a new connection: [Errno -2] Name or service not known'))
```
An nslookup works fine to the host mentioned in the error:
```
$ nslookup de.api.io.mi.com
Server:         127.0.0.53
Address:        127.0.0.53#53

Non-authoritative answer:
de.api.io.mi.com        canonical name = de-api-io-mi-com-new-1496482937.eu-central-1.elb.amazonaws.com.
Name:   de-api-io-mi-com-new-1496482937.eu-central-1.elb.amazonaws.com
Address: 3.122.23.174
Name:   de-api-io-mi-com-new-1496482937.eu-central-1.elb.amazonaws.com
Address: 3.120.212.222
Name:   de-api-io-mi-com-new-1496482937.eu-central-1.elb.amazonaws.com
Address: 18.159.80.136
Name:   de-api-io-mi-com-new-1496482937.eu-central-1.elb.amazonaws.com
Address: 18.194.167.186
```

**Comment by PiotrMachowski** [comment]
Fixed in v2.0.0

**Comment by ZuluWhiskey** [comment]
Just a quick question, the referenced PR mentions Viomi vacuums but I have an S5 Max, I just wanted to double check that it's the same?

**Comment by PiotrMachowski** [comment]
@ZuluWhiskey S5 Max should be supported for a quite some time

**Comment by ZuluWhiskey** [comment]
> @ZuluWhiskey S5 Max has been supported for a quite some time. Version v2.0.0 added support for Viomi and fixed some other bugs

Sorry, I'm slightly confused as I raised this issue as I'm getting the problem with my S5 Max? Apologies if I've missed something

**Comment by PiotrMachowski** [comment]
Yes, but generally S5 Max has been supported for a long time now. Your instance has encountered some random errors, which should have been fixed in new release (v2.0.0 contained a lot of changes, not all of them are Viomi-specific). Please try the latest version (v2.0.1) and check if it works for you.

**Comment by RaveGun** [comment]
In have updated and it works. There were no more errors in the log. For me the last update fixed the issue. Thank you

**Comment by ZuluWhiskey** [comment]
> Yes, but generally S5 Max has been supported for a long time now. Your instance has encountered some random errors, which should have been fixed in new release (v2.0.0 contained a lot of changes, not all of them are Viomi-specific). Please try the latest version (v2.0.1) and check if it works for you.

Brilliant, thanks for clearing that up! I'll update now ☺️
